src/js/component.js

```diff
@@ -568,36 +572,34 @@ Component.prototype.buildCSSClass = function(){
  *     myComponent.on(otherElement, 'eventName', myFunc);
  *     myComponent.on(otherComponent, 'eventName', myFunc);
  *
- * The benefit of using this over `vjs.on(otherElement, 'eventName', myFunc)`
+ * The benefit of using this over `Events.on(otherElement, 'eventName', myFunc)`
  * and `otherComponent.on('eventName', myFunc)` is that this way the listeners
  * will be automatically cleaned up when either component is disposed.
  * It will also bind myComponent as the context of myFunc.
  *
  * **NOTE**: When using this on elements in the page other than window
  * and document (both permanent), if you remove the element from the DOM
- * you need to call `vjs.trigger(el, 'dispose')` on it to clean up
+ * you need to call `Events.trigger(el, 'dispose')` on it to clean up
  * references to it and allow the browser to garbage collect it.
  *
- * @param  {String|vjs.Component} first   The event type or other component
+ * @param  {String|Component} first   The event type or other component
  * @param  {Function|String}      second  The event handler or event type
  * @param  {Function}             third   The event handler
- * @return {vjs.Component}        self
+ * @return {Component}        self
  */
 Component.prototype.on = function(first, second, third){
-  var target, type, fn, removeOnDispose, cleanRemover, thisComponent;
-
   if (typeof first === 'string' || Lib.obj.isArray(first)) {
     Events.on(this.el_, first, Lib.bind(this, second));
 
   // Targeting another component or element
   } else {
-    target = first;
-    type = second;
-    fn = Lib.bind(this, third);
-    thisComponent = this;
+    let target = first;
+    let type = second;
+    let fn = Lib.bind(this, third);
+    let thisComponent = this;
 
     // When this component is disposed, remove the listener from the other component
-    removeOnDispose = function(){
+    let removeOnDispose = ()=> {
```

Comment:
Here we could be using the arrow function it to replace `thisComponent`, assuming the IE8 issue mentioned above doesn't trump arrow functions all together. @dmlap, @gkatsev any thoughts on the IE8 named function issue? I've usually avoided using named functions when it's not needed, so can't speak to any experience running into the issue.